src/rabbit_reader.erl

```diff
@@ -1394,25 +1394,31 @@ auth_phase(Response,
                                        auth_mechanism = {Name, AuthMechanism},
                                        auth_state     = AuthState},
                        sock = Sock}) ->
+    Ip = list_to_binary(inet:ntoa(Connection#connection.host)),
     case AuthMechanism:handle_response(Response, AuthState) of
         {refused, Username, Msg, Args} ->
+            rabbit_core_metrics:auth_attempt_failed(Ip, Username),
```

Comment:
Should we track the protocol as well? In this case it would be `amqp091`.

Reply:
That's a great idea.